### PR TITLE
Update for AKS

### PR DESCRIPTION
**Pull request by milismsft**
update inners and add implementation to retrieve the Kubernetes cluster configuration settings (aka Kube.config)

### CODE REVIEW

File: src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs [hunk 1]
```diff
@@ -105,30 +110,50 @@ public string ServicePrincipalSecret()
...
+        {
+            if (this.userKubeConfigContent == null)
+            {
+                this.userKubeConfigContent = Extensions.Synchronize(() => this.GetUserKubeConfigContent());
+            }
+
+            return this.userKubeConfigContent;
+        }
+
+        private async Task<byte[]> GetAdminKubeConfigContent()
```
Comment (hovsepm) [inline src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs]: >GetAdminKubeConfigContent [](start = 35, length = 25)

Please rename this method to "*Async"

File: src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs [hunk 2]
```diff
@@ -105,30 +110,50 @@ public string ServicePrincipalSecret()
...
             else
             {
-                return System.Convert.FromBase64String(this.Inner.AccessProfiles.ClusterAdmin.KubeConfig);
+                return System.Convert.FromBase64String(profileInner.KubeConfig);
             }
         }
 
-        ///GENMHASH:94A9F49AECE7874D497A7F5E8A5AB25C:FB91DAFFC2B4B802AB4DE52634A023D9
-        public byte[] UserKubeConfigContent()
+        private async Task<byte[]> GetUserKubeConfigContent()
```
Comment (hovsepm) [inline src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs]: >GetUserKubeConfigContent [](start = 35, length = 24)

This one as well - please add Async suffix to the method name.

File: src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs (hunk 1 above)
Comment (milismsft) [reply src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs]: Fixed

File: src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs (hunk 2 above)
Comment (milismsft) [reply src/ResourceManagement/ContainerService/KubernetesClusterImpl.cs]: Fixed.

**Pull request merged**